### PR TITLE
Use installation method advised in otelcol builder README

### PR DESCRIPTION
**Pull request by albertteoh**
Signed-off-by: albertteoh <albert.teoh@logz.io>

Builds were configured to download a specific linux architecture's build of `opentelemetry-collector-builder` resulting in the following error when running `make` on my MacOS laptop:
```
$ make
/bin/sh: /Users/albertteoh/bin/opentelemetry-collector-builder: cannot execute binary file
make: *** [build-agent] Error 126
```

This PR proposes to follow the installation instructions documented in https://github.com/observatorium/opentelemetry-collector-builder:
```
go get github.com/observatorium/opentelemetry-collector-builder
```

which should ensure the resulting executable is compatible with the computer's OS + architecture.

## Testing

```
$ make
go get github.com/observatorium/opentelemetry-collector-builder@v0.2.0
2021-01-09T08:18:07.382+1100	INFO	cmd/root.go:82	OpenTelemetry Collector distribution builder	{"version": "dev", "date": "unknown"}
2021-01-09T08:18:07.383+1100	INFO	cmd/root.go:98	Using config file	{"path": "manifests/agent.yaml"}
2021-01-09T08:18:07.384+1100	INFO	builder/main.go:80	Sources created	{"path": "./builds/agent/"}
2021-01-09T08:18:07.385+1100	INFO	builder/main.go:94	Using go from PATH	{"Go executable": "/usr/local/bin/go"}
2021-01-09T08:18:07.385+1100	INFO	builder/main.go:97	Compiling
2021-01-09T08:18:08.004+1100	INFO	builder/main.go:103	Compiled	{"binary": "./builds/agent//jaeger-otel-agent"}
2021-01-09T08:18:08.015+1100	INFO	cmd/root.go:82	OpenTelemetry Collector distribution builder	{"version": "dev", "date": "unknown"}
2021-01-09T08:18:08.015+1100	INFO	cmd/root.go:98	Using config file	{"path": "manifests/collector.yaml"}
2021-01-09T08:18:08.017+1100	INFO	builder/main.go:80	Sources created	{"path": "./builds/collector/"}
2021-01-09T08:18:08.018+1100	INFO	builder/main.go:94	Using go from PATH	{"Go executable": "/usr/local/bin/go"}
2021-01-09T08:18:08.018+1100	INFO	builder/main.go:97	Compiling
2021-01-09T08:18:08.889+1100	INFO	builder/main.go:103	Compiled	{"binary": "./builds/collector//jaeger-otel-collector"}
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,5 @@
 module github.com/jaegertracing/jaeger-otelcol
 
 go 1.15
+
+require github.com/observatorium/opentelemetry-collector-builder v0.2.0 // indirect
```
Comment (yurishkuro) [inline go.mod:5]: How does this path get in here (and stay here)? I think the recommended convention is to define a dummy Go file with imports of the tools, like https://github.com/jaegertracing/jaeger/blob/master/tools.go

Comment (albertteoh) [reply go.mod]: Nice pickup @yurishkuro, you're right, a `go mod tidy` would remove this dependency.

I've added the dummy file and confirmed neither `go mod tidy` nor `make` modify the go.mod/go.sum files.

File: tools.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// +build tools
+
+package jaeger_otelcol
+
+// https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
+import (
+_ "github.com/observatorium/opentelemetry-collector-builder"
```
Comment (yurishkuro) [inline tools.go:7]: this doesn't look go-formatted, seems that CI does not run linter

Comment (albertteoh) [reply tools.go]: Copy-paste sloppiness on my part. I've added a `fmt` make target (copied from jaeger project) that should help prevent this.

How to enable linter in CI for this project? I'm happy to set it up if someone can point me in the right direction.

File: Makefile [hunk 3]
```diff
@@ -1,25 +1,45 @@
...
+				   -not -name 'gen_assets.go' \
+				   -not -name 'mocks*' \
+				   -not -name 'model.pb.go' \
+				   -not -name 'model_test.pb.go' \
+				   -not -name 'storage_test.pb.go' \
+				   -not -path './examples/*' \
+				   -not -path './vendor/*' \
+				   -not -path '*/mocks/*' \
+				   -not -path '*/*-gen/*' \
+				   -not -path '*/thrift-0.9.2/*' \
```
Comment (yurishkuro) [inline Makefile:18]: Are any of these exclusions actually relevant for this project? I would start with a minimum.

File: Makefile [hunk 4]
```diff
@@ -1,25 +1,45 @@
...
-else
-OTELCOL_BUILDER=$(shell which opentelemetry-collector-builder)
+	go get github.com/observatorium/opentelemetry-collector-builder@v$(OTELCOL_BUILDER_VERSION)
 endif
+OTELCOL_BUILDER=$(shell which opentelemetry-collector-builder)
+
+.PHONY: fmt
+fmt:
+	@echo Running go fmt on ALL_SRC ...
+	@$(GOFMT) -e -s -l -w $(ALL_SRC)
```
Comment (yurishkuro) [inline Makefile:45]: * I would define `lint` target and make it depend on `fmt`
* add a new step with `make lint` to https://github.com/jaegertracing/jaeger-otelcol/blob/2f0546350989122c1b5ed60afa3d7132ab5e2178/.github/workflows/ci.yaml#L24-L25

File: Makefile [hunk 5]
```diff
@@ -1,25 +1,49 @@
...
-OTELCOL_BUILDER_DIR ?= ~/bin
-OTELCOL_BUILDER ?= $(OTELCOL_BUILDER_DIR)/opentelemetry-collector-builder
+GOFMT=gofmt
+GOLINT=golint
+LINT_LOG=.lint.log
 
+# all .go files that are not auto-generated and should be auto-formatted and linted.
+ALL_SRC := $(shell find . -type d \( -name builds \) -prune -false -o \
+				   -name '*.go' \
+				   -not -name '.*' \
```
Comment (yurishkuro) [inline Makefile:9]: ?

File: tools.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+// +build tools
+
+package jaeger_otelcol
+
+// https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
+import (
+	_ "github.com/observatorium/opentelemetry-collector-builder"
```
Comment (yurishkuro) [inline tools.go:7]:
```suggestion
	_ "github.com/observatorium/opentelemetry-collector-builder"
	_ "golang.org/x/lint/golint"
```

**Review by yurishkuro — APPROVED** [review]
lgtm

File: Makefile (hunk 5 above)
Comment (albertteoh) [reply Makefile:9]: Excludes the `builds` directory which is generated by the opentelementry-collector-builder, as well as any "dot" files. Do you see something that needs fixing (or perhaps I can add a comment to help explain the `builds` exclusion)?

Comment (yurishkuro) [reply Makefile:9]: ok

**Comment by albertteoh** [comment]
@yurishkuro please let me know if there is anything more that needs addressing in this PR

**Review by yurishkuro — APPROVED** [review]

**Comment by albertteoh** [comment]
Thanks @yurishkuro!

**Comment by jpkrohling** [comment]
The installation instructions there are outdated, users are advised to download the binaries built during a release. This is because we might use built-time tags for the binary, which aren't available when `go get` is used.

File: tools.go [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
```
Comment (jpkrohling) [inline tools.go:7]: This change is a bit misleading and perhaps a bit dangerous: it implies that bumping the `go.mod` is sufficient, which isn't the case (yet?).

**Comment by albertteoh** [comment]
> The installation instructions there are outdated, users are advised to download the binaries built during a release. This is because we might use built-time tags for the binary, which aren't available when `go get` is used.

Thanks picking up on this, @jpkrohling. I can revert this back to downloading the binary (#20) instead. 

I've raised a minor issue relating to this, but it's not a blocker as I can always manually download and install it: https://github.com/observatorium/opentelemetry-collector-builder/issues/26.

File: tools.go (hunk 7 above)
Comment (albertteoh) [reply tools.go:7]: Okay, this can be removed as part of #20.

**Pull request merged**